.github/workflows/multi_nim_common.yml

```diff
@@ -0,0 +1,83 @@
+name: daily-common
+
+on:
+  workflow_call:
+    inputs:
+      nim-branch:
+        description: 'Nim branch'
+        required: true
+        type: string
+      platform:
+        description: 'Platform'
+        required: true
+        type: string
+
+jobs:
+  delete-cache:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: snnaplab/delete-branch-cache-action@v1
+
+  matrix-prep:
+
+  build:
+    needs: delete-cache
+    timeout-minutes: 120
+    strategy:
+      fail-fast: false
+      matrix:
+        target: ${{ fromJSON(inputs.platform) }}
+        branch: ${{ fromJSON(inputs.nim-branch) }}
+        include:
+          - target:
+              os: linux
+            builder: ubuntu-20.04
+            shell: bash
+          - target:
+              os: macos
+            builder: macos-12
+            shell: bash
+          - target:
+              os: windows
+            builder: windows-2019
+            shell: msys2 {0}
+
+    defaults:
+      run:
+        shell: ${{ matrix.shell }}
+
+    name: '${{ matrix.target.os }}-${{ matrix.target.cpu }} (Nim ${{ matrix.branch }})'
+    runs-on: ${{ matrix.builder }}
+    continue-on-error: ${{ matrix.branch == 'devel' || matrix.branch == 'version-2-0' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Nim
+        uses: "./.github/actions/install_nim"
+        with:
+          os: ${{ matrix.target.os }}
+          shell: ${{ matrix.shell }}
+          nim_branch: ${{ matrix.branch }}
+          cpu: ${{ matrix.target.cpu }}
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '~1.15.5'
+          cache: false
+
+      - name: Install p2pd
+        run: |
+          V=1 bash scripts/build_p2pd.sh p2pdCache 124530a3
+
+      - name: Run tests
+        run: |
+          nim --version
+          nimble --version
+          nimble install -y --depsOnly
```

Comment:
why aren't we using `nimble install_pinned` here? @lchenut @Menduist

Reply:
"nimble install_pinned" will be soon replaced by nimble.lock file which I have demonstrated in discord is working well. We will soon have new branch with other name, which will require some kind of a pinning to be later able to make regular releases.

Reply:
I would suggest we leave deps installation as it was previously. To pin or lock deps requires upfront testing, which at the current situation, with so many flaky tests, is not possible right now.

Reply:
> why aren't we using `nimble install_pinned` here? @lchenut @Menduist

I don't know why we are not using 'nimble install_pinned' in actual Daily workflow.

Reply:
The daily workflow is here to test regressions in our dependencies, so it always tests the latest versions of everything

Regarding nimble lock, it didn't exist back when we introduced the pinning system, but would indeed be good to migrate to it if it is stable enough. Also see https://github.com/status-im/nim-libp2p/pull/844

Reply:
It makes sense, but two points:

1.  What is the purpose of multi_nim_common.yml? It's not related to the regular tests in the PRs? Is it only for Daily?
2. Nimble `install` seems not to work properly and I am not sure removing all ranges is an acceptable strategy. See https://github.com/status-im/nim-bearssl/pull/52 and https://github.com/nim-lang/nimble/issues/1162.

Reply:
1. The purpose of multi_nim.yml ("Daily" workflow) is to cover mainstream platforms Linux, Mac, and Windows on amd64
2. Would it be acceptable for you we can discuss and try to identify "best shot" solution in upcoming PRs ?

Reply:
I think it is fine to merge it as it is, but it seems you need to sign your commits.

Reply:
I can sing new commits now, but signing old commits is not working for me yet. Do you have any suggestion what to about it ?

Reply:
I think you can squash the old commits

Reply:
Have you tried something like this https://superuser.com/questions/397149/can-you-gpg-sign-old-commits ?

Reply:
Thanks a lot for advice. I have tried, at the end I had to drop all unverified commits and submit all changes in the new commit. Please have a look.